Python/Product/Analysis/LanguageServer/Structures.cs

```diff
@@ -61,6 +61,13 @@ public struct Range {
         public static implicit operator Range(SourceSpan span) => new Range { start = span.Start, end = span.End };
 
         public override string ToString() => ((SourceSpan)this).ToString();
+
+        public Range Union(Range other) {
```

Comment:
I think changing the earlier struct to SourceSpan will make this easier, but I think it's better to keep operations like this on our own classes and only use the LSP structures for interop.